plugins/auth-backend-module-pinniped-provider/src/authenticator.test.ts

```diff
@@ -276,10 +299,133 @@ describe('pinnipedAuthenticator', () => {
               url: 'test',
             },
           } as unknown as OAuthAuthenticatorStartInput,
-          implementation,
+          authCtx,
         ),
       ).rejects.toThrow('authentication requires session support');
     });
+
+    it('refreshes oidc metadata after a failed fetch', async () => {
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, _ctx) => res.networkError('Timeout'),
+        ),
+      );
+
+      const authCtxCreatedWhileSupervisorUnavailable =
+        pinnipedAuthenticator.initialize({
+          callbackUrl: 'https://backstage.test/callback',
+          config: new ConfigReader({
+            federationDomain: 'https://federationDomain.test',
+            clientId: 'clientId',
+            clientSecret: 'clientSecret',
+          }),
+        });
+
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, ctx) =>
+            res(
+              ctx.status(200),
+              ctx.set('Content-Type', 'application/json'),
+              ctx.json(issuerMetadata),
+            ),
+        ),
+      );
+
+      const response = await pinnipedAuthenticator.start(
+        startRequest,
+        authCtxCreatedWhileSupervisorUnavailable,
+      );
+
+      expect(response.url).toMatch('https://pinniped.test/oauth2/authorize');
+    });
+
+    it('caches oidc metadata after a success', async () => {
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, _ctx) => res.networkError('Timeout'),
+        ),
+      );
+
+      const authCtxCreatedWhileSupervisorUnavailable =
+        pinnipedAuthenticator.initialize({
+          callbackUrl: 'https://backstage.test/callback',
+          config: new ConfigReader({
+            federationDomain: 'https://federationDomain.test',
+            clientId: 'clientId',
+            clientSecret: 'clientSecret',
+          }),
+        });
+
+      let supervisorCalls: number = 0;
+
+      mswServer.use(
+        rest.get(
+          'https://federationDomain.test/.well-known/openid-configuration',
+          (_req, res, ctx) => {
+            supervisorCalls += 1;
+            return res(
+              ctx.status(200),
+              ctx.set('Content-Type', 'application/json'),
+              ctx.json(issuerMetadata),
+            );
+          },
+        ),
+      );
+
+      await pinnipedAuthenticator.start(
+        startRequest,
+        authCtxCreatedWhileSupervisorUnavailable,
```

Comment:
If you agree with my above comment, then you can use the `beforeEach` for this:
```suggestion
        authCtx,
```

Reply:
Refactored and changed.